spacy/matcher/dependencymatcher.pyx

```diff
@@ -179,29 +176,22 @@ cdef class DependencyMatcher:
         self._callbacks[key] = on_match
 
         # Add 'RIGHT_ATTRS' to self._patterns[key]
-        _patterns = []
-        for pattern in patterns:
-            token_patterns = []
-            for i in range(len(pattern)):
-                token_pattern = [pattern[i]["RIGHT_ATTRS"]]
-                token_patterns.append(token_pattern)
-            _patterns.append(token_patterns)
+        _patterns = [[[pat["RIGHT_ATTRS"]] for pat in pattern] for pattern in patterns]
         self._patterns[key].extend(_patterns)
 
         # Add each node pattern of all the input patterns individually to the
         # matcher. This enables only a single instance of Matcher to be used.
         # Multiple adds are required to track each node pattern.
         tokens_to_key_list = []
-        for i in range(len(_patterns)):
+        for i, current_patterns in enumerate(_patterns):
 
             # Preallocate list space
-            tokens_to_key = [None]*len(_patterns[i])
+            tokens_to_key = [None] * len(current_patterns)
 
             # TODO: Better ways to hash edges in pattern?
-            for j in range(len(_patterns[i])):
-                k = self._get_matcher_key(key, i, j)
-                self._matcher.add(k, [_patterns[i][j]])
-                tokens_to_key[j] = k
+            for j, _pattern in enumerate(current_patterns):
+                tokens_to_key[j] = k = self._get_matcher_key(key, i, j)
+                self._matcher.add(k, [_pattern])
```

Comment:
I'm not sure we use this kind of double assignment anywhere else, so for readability I'd just keep it at
```suggestion
                k = self._get_matcher_key(key, i, j)
                self._matcher.add(k, [_pattern])
                tokens_to_key[j] = k
```